src/quacc/recipes/espresso/phonons.py

```diff
@@ -56,10 +81,180 @@ def phonon_job(
+
+    1. pw.x relaxation
+        - name: "pw_job"
+        - job: [quacc.recipes.espresso.core.relax_job][]
+
+    2. ph.x calculation test_run
+        - name: "ph_test_job"
+        - job: [quacc.recipes.espresso.phonons.phonon_job][]
+
+    3. (n * m) / nblocks ph.x calculations
+        - name: "ph_job"
+        - job: [quacc.recipes.espresso.phonons.phonon_job][]
+
+    4. ph.x calculation to gather data and diagonalize each dynamical matrix
+        - name: "ph_recover_job"
+        - job: [quacc.recipes.espresso.phonons.phonon_job][]
+
+    Parameters
+    ----------
+    atoms
+        Atoms object
+    nblocks
+        The number of representations to group together in a single job.
+        This will reduce the amount of data produced by a factor of nblocks.
+        If nblocks = 0, each job will contain all the representations for a
+        single q-point.
+    job_params
+        Custom parameters to pass to each Job in the Flow. This is a dictinoary where
+        the keys are the names of the jobs and the values are dictionaries of parameters.
+    job_decorators
+        Custom decorators to apply to each Job in the Flow. This is a dictionary where
+        the keys are the names of the jobs and the values are decorators.
+
+    Returns
+    -------
+    RunSchema
+        Dictionary of results from [quacc.schemas.ase.summarize_run][]
+    """
+
+    # Prepare the job parameters and decorators
+    calc_defaults = {
+        "pw_job": {
+            "input_data": {
+                "control": {"forc_conv_thr": 5.0e-5},
+                "electrons": {"conv_thr": 1e-12},
+            }
+        }
+    }
+    job_params = recursive_dict_merge(calc_defaults, job_params)
+    pw_job, ph_test_job, ph_job, recover_ph_job = customize_funcs(
+        ["pw_job", "ph_test_job", "ph_job", "ph_recover_job"],
+        [relax_job, phonon_job, phonon_job, phonon_job],
+        parameters=job_params,
+        decorators=job_decorators,
+    )
+
+    # Get the ph.x input_data from the input job parameters
+    ph_job_input_data = job_params.get("ph_job", {}).get("input_data", {})
+
+    # Run the pw.x relaxation
+    pw_job_results = pw_job(atoms)
+
+    # Run the cheap ph.x test calculation
+    ph_test_job_results = ph_test_job(
+        pw_job_results["dir_name"], test_run=True, input_data=ph_job_input_data
+    )
+
+    # Run the grid ph.x subflow
+    grid_results = _grid_phonon_subflow(
+        ph_job_input_data,
+        ph_test_job_results["dir_name"],
+        pw_job_results["dir_name"],
+        ph_job,
+        nblocks=nblocks,
+    )
+
+    # Prep for the final recover ph.x job
+    copy_back = [result["dir_name"] for result in grid_results]
+
+    input_data = grid_results[-1]["parameters"]["input_data"]
+    for k in ["start_q", "last_q", "start_irr", "last_irr"]:
+        input_data["inputph"].pop(k)
+    input_data["inputph"]["recover"] = True
+
+    return recover_ph_job(copy_back, input_data=input_data)
```

Comment:
New commits should fix that by changing the whole thing simply to

``` python
ph_job_input_data["recover"] = True
```

Which makes even more sense because the same base ph_input_data should be used everywhere. Although I am a little bit in the future because this requires the flat dict come-back.